```{r global_options, message = FALSE}
library(knitr) # enables knit output formatting
  opts_chunk$set(fig.align="center", fig.height=3, fig.width=4) # format knit output
library(tidyverse) # contains ggplot2, dplyr, tidyr, readr, stringr, etc
  theme_set(theme_bw(base_size = 12)) # set default plot theme for ggplot2
library(stringr) # string manipulation functions
library(ggrepel) # enables dynamic plot labeling
library(ggthemes) # enables colorblind-friendly color palette
library(scales) # imports function show_col() for displaying color palettes
library(janitor) # imports function clean_names() for auto-formatting bad column names
library(ggbreak) # for truncating axes with pesky outliers
```

```{r message = FALSE, fig.height=5, fig.width=8}
# custom qualitative color palettes
palette_npg <- c("#E64B35", "#4DBBD5", "#00A087", "#3C5488",
                 "#F39B7F", "#8491B4", "#91D1C2", "#DC0000",
                 "#7E6148", "#B09C85")
pal_clades <- c("#E64B35", "#4DBBD5", "#3C5488", "#00A087")
workdir <- "/stor/work/Marcotte/project/rmcox/leca/"
```



# Generate heat maps for LECA OG coverage for each collapsed eukaryotic  proteome in the CFMS data
**01/04/23**

------

This document contains research notes and code generated by:

> **Rachael M. Cox**  
> Biochemistry PhD Candidate  
> University of Texas at Austin  
> rachaelcox@utexas.edu

------

### Read in & format data

```{r}
# coverage data
cov <- read_csv('ppi_ml/results/coverage/og_coverage_by_species_phylo.150p.filtdollo.csv')

# species names & clades
species <- read_csv('ppi_ml/data/meta/speciesinfo_clades.csv') %>%
  filter(tax_group == "eukaryota") %>% 
  select(code, species_name, clade) %>%
  mutate(code = tolower(code))

# ordered codes by clade -> alphabetical
amor <- read_delim('ppi_ml/data/meta/amorphea_codes_ordered.txt', delim='\n',
                   col_names=FALSE) %>% pull(X1)
exca <- read_delim('ppi_ml/data/meta/excavate_codes_ordered.txt', delim='\n',
                   col_names=FALSE) %>% pull(X1)
tsar <- read_delim('ppi_ml/data/meta/tsar_codes_ordered.txt', delim='\n',
                   col_names=FALSE) %>% pull(X1)
viri <- read_delim('ppi_ml/data/meta/viridiplantae_codes_ordered.txt', delim='\n',
                   col_names=FALSE) %>% pull(X1)

# ordered codes based on phylogeny
tree <- read_delim('ppi_ml/data/meta/euk_codes_ordered_phylo.txt',
                   delim='\n', col_names=FALSE) %>% pull(X1)

# clade order
clade_order <- c('Amorphea','Excavate','TSAR','Archaeplastida')

# og annotations
annots <- read_tsv('ppi_ml/annotations/leca_euNOGs_annotated.fmt.tsv')

annots_fmt <- annots %>% 
  select(ID, matches('*genes_fmt*'),
         human_protein_names,
         human_function_cc) %>%
  mutate(gene_names = coalesce(human_genes_fmt,
                              arath_genes_fmt,
                              ID)) %>%
  select(-matches('*genes_fmt*')) %>%
  select(ID, gene_names, everything())

# reorder coverage based on clusters
clst_order <- cov %>%
  left_join(annots_fmt, by=c("ID")) %>%
  select(ID, gene_names) %>%
  unique()

# ugggggggggghhhhhhhhhhh there are 18 kogs with dup gene names
# 15 of these are a single uniprot ID mapping to multiple ogs
# TODO: fix this somehow
dups <- clst_order %>%
  group_by(gene_names) %>%
  tally() %>% 
  arrange(desc(n)) %>%
  filter(n > 1)
```

```{r}

```

### Plot protein family size per orthogroup for each eukaryote in the data set

```{r}
`%!in%` = Negate(`%in%`)

# fam sizes for multiple euks 
# + mean/median over entire data set 
avg_sizes <- read_csv(paste0(workdir, "ppi_ml/data/meta/avg_famsize_per_og.csv")) %>%
  filter(ID %in% leca_ogs)

missing_prots <- leca_ogs[leca_ogs %!in% avg_sizes$ID]
missing_ms <- leca_ogs[leca_ogs %!in% cov$ID]

# reorder
ordered_cols <- avg_sizes[tree] 
avg_ord <- avg_sizes %>%
  select(ID) %>%
  cbind(ordered_cols)

# make subset
avg_subset <- avg_sizes %>%
  select(ID, brart, human, euggr, tetts, 
         chlre, arath, matches("*family_size*")) %>%
  pivot_longer(brart:arath, names_to = "species",
               values_to = "family_size")
```

Correlation plot; not my favorite.
```{r}
# plot correlation for subet; not sure I like this ...
ggplot(avg_subset, aes(x = median_family_size, 
                      y = family_size,
                      color = species)) +
  geom_point(size = 2, alpha = 0.75) +
  scale_color_manual(values = pal_npg) +
  labs(x = "Median # proteins per orthogroup\nacross all eukaryotes",
       y = "Actual # proteins per orthogroup") +
  facet_wrap(~species) +
  theme(legend.position = "none")


# save output
figure_dir <- paste0(workdir, "ppi_ml/figures/")
ggsave(paste0(figure_dir, "famsize_per-og_per-species_unscaled.png"),
       device = "png", width = 6, height = 6, units = "in")
```

Boxplot looks nice though:
```{r}
# format data long
avg_ordl <- avg_ord %>%
  pivot_longer(dicdi:soybn, names_to = "species",
               values_to = "num_prots")

# get num OGs
num_ogs <- length(avg_ord$ID)

# lock in var order
avg_ordl$species <- factor(avg_ordl$species, levels = tree)

# boxplot (I like this much better)
avg_ordl %>%
  left_join(species, by = c("species" = "code")) %>%
  mutate(species_name_fmt = str_replace(
    species_name, '(\\s)(?=\\()', '\n')) %>% 
  ggplot(., aes(x = fct_relevel(species, tree),
                y = num_prots, 
                fill = fct_relevel(clade, clade_order))) +
    geom_boxplot() +
    scale_fill_manual(values = pal_clades) +
    scale_y_log10() +
    #scale_y_break(c(3,10)) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          legend.title = element_blank(),
          legend.position = "top") +
    labs(x = "Species", y = "Number of proteins per orthogroup\n(5988 orthogroups, 150 PSM threshold)")

# save plot
figure_dir <- paste0(workdir, "ppi_ml/figures/")
ggsave(paste0(figure_dir, "famsize_per-og_per-species_boxplot_fd_150p.png"),
       device = "png", width = 6, height = 6, units = "in")
```

### Plot characteristics of missing data

What are we **not** measuring?

```{r}
`%!in%` = Negate(`%in%`)

# fam sizes for multiple euks 
# + mean/median over entire data set 
avg_sizes <- read_csv(paste0(workdir, "ppi_ml/data/meta/avg_famsize_per_og.csv"))

# CFMS coverage
cov <- read_csv('ppi_ml/results/coverage/og_coverage_by_species.0p.filtdollo.csv')

# all leca OGs
leca_ogs <- pull(annots, ID) %>% unique()

# missing observations
# what's happening here?
missing_prots <- leca_ogs[leca_ogs %!in% avg_sizes$ID] 
missing_ms <- leca_ogs[leca_ogs %!in% cov$ID]

length(intersect(missing_prots, missing_ms)) # 2337 overlap
```

### Plot heat maps

All LECA (filtdollo) data:

```{r}
# coverage data
thres = 0
cov_file = str_glue('ppi_ml/results/coverage/og_coverage_by_species_phylo.{thres}p.filtdollo.csv')
cov <- read_csv(cov_file)

# get num OGs
num_ogs <- length(cov$ID)

# tidy data & join annots
cov_tidy <- cov %>%
  pivot_longer(!ID, names_to = "species", values_to = "presence") %>%
  left_join(species, by=c("species" = "code")) %>%
  left_join(annots_fmt, by=c("ID"))

# lock in var orders
cov_tidy$ID <- factor(cov_tidy$ID, levels = cov$ID)

cov_tidy$clade <- factor(cov_tidy$clade, levels = clade_order)

species_order <- c(amor, exca, tsar, viri)
cov_tidy$species <- factor(cov_tidy$species, levels = species_order)

#theme_set(cowplot::theme_cowplot()) # ugly
theme_set(theme_bw())
ggplot(cov_tidy, aes(x = species, y = fct_rev(ID), 
                     fill = clade, alpha = as.factor(presence))) +
  geom_tile(linejoin = "round") +
  scale_fill_manual(values = pal_clades) +
  theme(axis.text.x = element_text(angle = 45, 
                                   vjust = 1, 
                                   hjust = 1),
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.title.x = element_blank(),
        #axis.title.y = element_blank(),
        legend.position = "top",
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_blank(),
        legend.key.size = unit(0.25, 'cm'),
        axis.line.y=element_blank()) +
  #geom_segment(aes(x=0, xend=0, y=num_ogs, yend=0), size=1, arrow = arrow(length = unit(0.6,"cm"))) +
  labs(y = str_glue("{num_ogs} orthogroups with >= {thres} PSMs"), 
       x = "", fill = "", alpha = "") +
  coord_cartesian(clip = "off")

outname <- str_glue('ppi_ml/figures/og_coverage_by_species_phylo.{thres}p.filtdollo.png')
ggsave(outname, device = "png", 
       width = 6, height = 6, units = "in",
       dpi = 300)
```

All LECA (filtdollo) data **with gene names**:

```{r}
# coverage data
thres = 0
cov_file = str_glue('ppi_ml/results/coverage/og_coverage_by_species_phylo.{thres}p.filtdollo.csv')
cov <- read_csv(cov_file)

# get num OGs
num_ogs <- length(cov$ID)

# tidy data & join annots
cov_tidy <- cov %>%
  pivot_longer(!ID, names_to = "species", values_to = "presence") %>%
  left_join(species, by=c("species" = "code")) %>%
  left_join(annots_fmt, by=c("ID"))

# lock in var orders
cov_tidy$ID <- factor(cov_tidy$ID, levels = cov$ID)

cov_tidy$clade <- factor(cov_tidy$clade, levels = clade_order)

species_order <- c(amor, exca, tsar, viri)
cov_tidy$species <- factor(cov_tidy$species, levels = species_order)

#theme_set(cowplot::theme_cowplot()) # ugly
theme_set(theme_bw())
ggplot(cov_tidy, aes(x = species, y = fct_rev(ID), 
                     fill = clade, 
                     alpha = as.factor(presence))) +
  geom_tile(linejoin = "round") +
  scale_fill_manual(values = pal_clades) +
  #scale_y_discrete(expand = c(-1,0.2)) +
  #scale_x_discrete(expand = c(0.01,0)) +
  theme(axis.text.x = element_text(angle = 45, 
                                   vjust = 1, 
                                   hjust = 1),
        axis.text.y = element_text(size=2),
        axis.ticks.y = element_blank(),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        legend.position = "top",
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_blank(),
        legend.key.size = unit(0.25, 'cm'),
        axis.line.y=element_blank()) +
  #geom_segment(aes(x=0, xend=0, y=num_ogs, yend=0), size=1, arrow = arrow(length = unit(0.6,"cm"))) +
  labs(y = "", 
       x = "", fill = "", alpha = "") +
  coord_cartesian(clip = "off")

outname <- str_glue('ppi_ml/figures/og_coverage_by_species_phylo_w-genes.{thres}p.filtdollo.png')
ggsave(outname, device = "png", 
       width = 6, height = 18, units = "in")
```

### Plot dot plots

Test code for 1 complex (LSM spliceosome):
```{r}
# complex
cmplx_name = "LSM spliceosome"
cmplx = c("KOG1782", "KOG1784", "KOG1783", "KOG3448", "KOG1781", "KOG1775", "KOG3293", "KOG3460")
num_units = length(cmplx)

# coverage data
thres = 150
cov_file = str_glue('ppi_ml/results/coverage/og_coverage_by_species_phylo.{thres}p.filtdollo.csv')
cov_cmplx <- read_csv(cov_file) %>%
  filter(ID %in% cmplx)

# tidy data & join annots
cov_tidy <- cov_cmplx %>%
  pivot_longer(!ID, names_to = "species", values_to = "presence") %>%
  left_join(species, by=c("species" = "code")) %>%
  left_join(annots_fmt, by=c("ID"))

# tidy data & join annots
cov_tidy <- cov_cmplx %>%
  pivot_longer(!ID, names_to = "species", values_to = "presence") %>%
  left_join(species, by=c("species" = "code")) %>%
  left_join(annots_fmt, by=c("ID"))

# lock in var orders
cov_tidy$ID <- factor(cov_tidy$ID, levels = cov$ID)
cov_tidy$clade <- factor(cov_tidy$clade, levels = clade_order)
cov_tidy$species <- factor(cov_tidy$species, levels = tree)

# dot plot
ggplot(cov_tidy, aes(x = species, y = fct_rev(gene_names), color = clade,
                     )) +
  geom_point(shape = 1, color = "black") +
  geom_point(aes(color = clade, 
                 size = ifelse(presence==0, NA, 2.5))) +
  #geom_point(aes(color = clade)) +
  theme_bw() +
  scale_color_manual(values = pal_clades) +
  theme(axis.text.x = element_text(angle = 45, 
                                   vjust = 1, 
                                   hjust = 1),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        legend.position = "bottom",
        legend.margin=margin(t=-10)) +
  scale_size(guide = 'none') +
  labs(color = "", title = cmplx_name)

# save
name_fmt <- str_replace(cmplx_name, ' ', '-')
outname <- str_glue('ppi_ml/figures/cmplx_members_by_species_{name_fmt}.png')
ggsave(outname, device = "png", 
       width = 6, height = 3, units = "in",
       dpi = 300)
```

## Complex lists:
**LSM spliceosome**
cmplx_name = "LSM spliceosome"
cmplx = c("KOG1782", "KOG1784", "KOG1783", "KOG3448", "KOG1781", "KOG1775", "KOG3293", "KOG3460")

**Uncharacterized metabolic pathway complex**
cmplx_name = "Uncharacterized metabolic pathway complex"
cmplx = c("ENOG502S79E", "ENOG502S7PR", "ENOG502RZ3J", "ENOG502S2V0")
